js/bitbay.js

```diff
@@ -140,6 +141,164 @@ module.exports = class bitbay extends Exchange {
+            const quoteId = this.safeString (nativeMarket['second'], 'currency');
+            const id = baseId + quoteId;
+            const base = this.commonCurrencyCode (baseId);
+            const quote = this.commonCurrencyCode (quoteId);
+            const precision = this.safeInteger (nativeMarket['second'], 'scale');
+            // todo: check that the limits have ben interpreted correctly
+            // todo: parse the fees page
+            result.push ({
+                'id': id,
+                'symbol': base + '/' + quote,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'precision': precision,
+                'active': undefined,
+                'fee': undefined,
+                'limits': {
+                    'amount': {
+                        'min': this.safeFloat (nativeMarket['first'], 'minOffer'),
+                    },
+                    'cost': {
+                        'min': this.safeFloat (nativeMarket['second'], 'minOffer'),
+                    },
+                },
+                'info': item,
+            });
+        }
+        return result;
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = this.extend ({
+            'markets': symbol ? [this.marketId (symbol)] : [],
+        }, params);
+        const response = await this.v1_01PrivateGetTradingHistoryTransactions ({ 'query': this.json (request) });
+        if (response['status'] !== 'Ok')
+            throw new ExchangeError (this.id + ' balances query failed ' + this.json (response));
+        //   { status: 'Ok',
+        //     totalRows: '67',
+        //     items:
+        //     [ { id: 'b54659a0-51b5-42a0-80eb-2ac5357ccee2',
+        //         market: 'BTC-EUR',
+        //         time: '1541697096247',
+        //         amount: '0.00003',
+        //         rate: '4341.44',
+        //         initializedBy: 'Sell',
+        //         wasTaker: false,
+        //         userAction: 'Buy',
+        //         offerId: 'bd19804a-6f89-4a69-adb8-eb078900d006',
+        //         commissionValue: null }, ...
+        const items = response['items'];
+        const result = this.parseTrades (items, undefined, since, limit);
+        if (symbol === undefined)
+            return result;
+        return this.filterBySymbol (result, symbol);
+    }
+
+    async fetchTransactions (code = undefined, since = undefined, limit = undefined, params = {}) {
+        const request = this.extend ({
+            'balanceCurrencies': code ? [this.commonCurrencyCode (code)] : [],
+            'fromTime': since,
+            'limit': limit,
+        }, params);
+        const response = await this.v1_01PrivateGetBalancesBITBAYHistory ({ 'query': this.json (request) });
```

Comment:
This is practically the same story with `fetchLedger` as with the other exchanges.